indico/modules/events/surveys/__init__.py

```diff
@@ -55,18 +55,29 @@ def _visible(event):
     return EventMenuEntry('survey.display_survey_list', _('Surveys'), name='surveys', visible=_visible)
 
 
-@template_hook('event-header')
-def _inject_event_header(event, **kwargs):
+def _get_active_surveys(event):
     from indico.modules.events.surveys.models.surveys import Survey
```

Comment:
```
if not event.has_feature('surveys'):
    return []
```

(then disable the "surveys" feature and check that the survey announcement doesn't show up anymore)